src/common/services/api/order.service.js

```diff
@@ -316,6 +380,93 @@ class Order {
       return startedOrderWithoutSpouse
     }
   }
+
+  calculatePricesWithFees (feesApplied, cartItems) {
+    angular.forEach(cartItems, (item) => {
+      if (feesApplied) {
+        item.amountWithFee = item.amount
+        item.priceWithFee = item.price
+      } else {
+        item.amountWithFee = round(this.calculateAmountWithFees(item.amount), 2)
+        item.priceWithFee = this.calculatePriceWithFees(item.amount)
+      }
+    })
+    return true
+  }
+
+  calculatePriceWithFees (originalAmount) {
+    const newAmount = this.calculateAmountWithFees(originalAmount)
+    return `$${this.$filter('number')(newAmount, 2)}`
+  }
+
+  calculateAmountWithFees (originalAmount) {
+    originalAmount = parseFloat(originalAmount)
+    return originalAmount / this.FEE_DERIVATIVE
+  }
+
+  updatePrices (cartData) {
+    this.storeCoverFeeDecision(cartData.coverFees)
+    let newTotal = 0
+
+    angular.forEach(cartData.items, (item) => {
+      let newAmount
+      if (cartData.coverFees) {
+        newAmount = item.amountWithFee
+      } else {
+        if (parseFloat(item.amount) === parseFloat(item.amountWithFee)) {
+          newAmount = this.calculateAmountWithoutFees(item.amount)
+        } else {
+          newAmount = item.amount
+        }
+      }
+
+      item.amount = round(parseFloat(newAmount), 2)
+      item.config.amount = item.amount
+      item.price = `$${this.$filter('number')(newAmount, 2)}`
+      newTotal += item.amount
+    })
+
+    cartData.cartTotal = newTotal
+    this.recalculateFrequencyTotals(cartData)
+    this.storeCartData(cartData)
+  }
+
+  recalculateFrequencyTotals (cartData) {
+    angular.forEach(cartData.frequencyTotals, rateTotal => {
+      rateTotal.total = '$0.00'
+      rateTotal.amount = 0
+    })
+
+    angular.forEach(cartData.items, item => {
+      angular.forEach(cartData.frequencyTotals, rateTotal => {
+        if (item.frequency === rateTotal.frequency) {
+          rateTotal.amount += item.amount
+          rateTotal.total = `$${this.$filter('number')(rateTotal.amount, 2)}`
+        }
+      })
+    })
+  }
+
+  calculatePriceWithoutFees (originalAmount) {
+    const newAmount = this.calculateAmountWithoutFees(originalAmount)
+    return this.$filter('number')(newAmount, 2)
+  }
+
+  calculateAmountWithoutFees (originalAmount) {
+    originalAmount = parseFloat(originalAmount)
+    return originalAmount * this.FEE_DERIVATIVE
+  }
+
+  editGifts (cartData) {
+    const observables = []
+    angular.forEach(cartData.items, item => {
```

Comment:
`cartData.items.map(...)` is probably cleaner than a `forEach` and a `push`. As long as `cartData.items` is always an array. Otherwise you could do `(cartData.items || []).map(...)`